os/drivers/lcd/st7785_st7701.c

```diff
@@ -0,0 +1,420 @@
+ *
+ *  dev     - The lcd device
+ *  row     - Starting row to read from (range: 0 <= row < yres)
+ *  col     - Starting column to read read (range: 0 <= col <= xres-npixels)
+ *  buffer  - The buffer in which to return the run read from the LCD
+ *  npixels - The number of pixels to read from the LCD
+ *            (range: 0 < npixels <= xres-col)
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_LCD_NOGETRUN
+static int lcd_getrun(FAR struct lcd_dev_s *dev, fb_coord_t row, fb_coord_t col, FAR uint8_t *buffer, size_t npixels)
+{
+	return OK;
+}
+#endif
+
+/****************************************************************************
+ * Name:  lcd_getvideoinfo
+ *
+ * Description:
+ *   Get information about the LCD video controller configuration.
+ *
+ ****************************************************************************/
+
+static int lcd_getvideoinfo(FAR struct lcd_dev_s *dev, FAR struct fb_videoinfo_s *vinfo)
+{
+	DEBUGASSERT(dev && vinfo);
+	//vinfo->fmt = st7785_COLORFMT; /* Color format: RGB16-565: RRRR RGGG GGGB BBBB */
+	vinfo->xres = XRES;	/* Horizontal resolution in pixel columns */
+	vinfo->yres = YRES;	/* Vertical resolution in pixel rows */
+	vinfo->nplanes = MAX_NO_PLANES;	/* Number of color planes supported */
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_getplaneinfo
+ *
+ * Description:
+ *   Get information about the configuration of each LCD color plane.
+ *
+ ****************************************************************************/
+
+static int lcd_getplaneinfo(FAR struct lcd_dev_s *dev, unsigned int planeno, FAR struct lcd_planeinfo_s *pinfo)
+{
+	DEBUGASSERT(dev && pinfo && planeno == 0);
+	pinfo->putrun = &lcd_putrun;	/* Put a run into LCD memory */
+	pinfo->putarea = &lcd_putarea;	/* Put an area into LCD */
+#ifndef CONFIG_LCD_NOGETRUN
+	pinfo->getrun = &lcd_getrun;	/* Get a run from LCD memory */
+#endif
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_getpower
+ ****************************************************************************/
+
+static int lcd_getpower(FAR struct lcd_dev_s *dev, int color)
+{
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_setpower
+ ****************************************************************************/
+
+static int lcd_setpower(FAR struct lcd_dev_s *dev, int power)
+{
+	FAR struct st7785_st7701_dev_s *priv = (FAR struct st7785_st7701_dev_s *)dev;
+	if (power > CONFIG_LCD_MAXPOWER) {
+		printf("Power exceeds CONFIG_LCD_MAXPOWER %d", CONFIG_LCD_MAXPOWER);
+		return -1;
+	}
+	if (power > 0) {
+                priv->config->backlight(power);
+                priv->power = power;
+        } else {
+		priv->config->backlight(power);
+                priv->power = 0;
+        }
```

Comment:
1. How about below to reduce duplicated code?
```
if (power < 0) {
     power = 0;
}
priv->power = power;
priv->config->backlight(priv->power);
```

2. os/driver should follow Tizen Lite coding rules. Indentation should be a tab, not space.